src/ansys/fluent/core/solver/settings_builtin_bases.py

```diff
@@ -3,44 +3,86 @@
 from typing import Optional
 
 from ansys.fluent.core.session_solver import Solver
+from ansys.fluent.core.solver.flobject import SettingsBase
 from ansys.fluent.core.solver.settings_builtin_data import DATA
 
 
+def _get_obj(solver: Solver, cls_name: str):
+    if not isinstance(solver, Solver):
+        raise TypeError(f"{solver} is not a Solver object.")
+    path = DATA[cls_name][1]
+    obj = solver.settings
+    if isinstance(path, dict):
+        version = solver.get_fluent_version()
+        path = path.get(version)
+        if path is None:
+            raise RuntimeError(
+                f"{cls_name} is not supported in Fluent version {version}."
+            )
+    for comp in path.split("."):
+        obj = SettingsBase.__getattribute__(obj, comp)  # bypass InactiveObjectError
+    return obj
+
+
 class _SingletonSetting:
-    def __new__(cls, solver: Solver):
-        obj = solver.settings
-        path = DATA[cls.__name__][1]
-        if isinstance(path, dict):
-            version = solver.get_fluent_version()
-            path = path.get(version)
-            if path is None:
-                raise RuntimeError(
-                    f"{cls.__name__} is not supported in Fluent version {version}."
-                )
-        for comp in path.split("."):
-            obj = getattr(obj, comp)
-        return obj
-
-
-class _CreatableNamedObjectSetting(_SingletonSetting):
-    def __new__(
-        cls,
-        solver: Solver,
+    def __init__(self, solver: Optional[Solver] = None):
+        self.__dict__.update(dict(solver=None))
+        if solver is not None:
+            self.solver = solver
+
+    def __setattr__(self, name, value):
+        if name == "solver":
+            obj = _get_obj(value, self.__class__.__name__)
+            self.__class__ = obj.__class__
+            self.__dict__.clear()
```

Comment:
Clearing existing attributes (`solver`/`name`) as they may conflict with settings child of same name.

Reply:
Whenever user provides a `solver` object, should we be flexible and allow this to be either:
(a) a `settings` object (`<solver-session>.settings`), or
(b) a `<solver-session>` object

We would check for **(a)** first via a type check. Then we check for **(b)** by looking for a `settings` attribute. In both cases, we would store the `settings` object. Presumably, there's no broader dependency on the solver.

Reply:
@seanpearsonuk I'll add that. One problem with storing the attribute is that some settings classes have child named `solver` or `settings`. So the attribute name will conflict with those. In the current code, I've cleared the existing attributes before casting it to the settings object.

Agree that the following behaviour is non-intuitive:
```
x.solver = solver
x.solver  # AttributeError
```

Reply:
I've changed the attribute name to `settings_source`. The settings API doesn't have any settings class with that name, so that attribute can now be stored in the transformed settings object under that name.

Also, the dependency of solver/session is removed from builtin class implementation.